test/test_query_geo.cpp

```diff
@@ -180,4 +187,66 @@ TEST(Geospatial_GeoWithinShapes)
     }
 }
 
+TEST(Geospatial_PolygonValidation)
```

Comment:
Not sure if those cases are already covered in other tests, but a few more invalid polygons I can think of are:

1. Polygon with duplicate points: `[{0,0}, {0,1}, {0,1}, {1,1}, {0,0}]`
2. Polygon that intersects itself: `[{0,0}, {0,1}, {2,1}, {2,2}, {0,0}]`
3. Polygon that is a line: `[{0,0}, {1,1}, {2,2}, {0,0}]`
4. Polygon where the holes intersect with one another
5. Validation that CCW and CW polygon definitions give the same result

Reply:
1. This is valid because the prefiltering removes duplicate adjacent points. Added a test for this.
2. Added a test.
3. This polygon is valid according to S2, I think this is because of the mapping of points to a sphere.
4. There was already a test for intersecting holes.
5. Added a test for this.

Thanks for the test ideas!